src/panels/lovelace/cards/energy/hui-energy-usage-graph-card.ts

```diff
@@ -662,7 +662,10 @@ export class HuiEnergyUsageGraphCard
       });
     });
 
-    const uniqueKeys = Array.from(new Set(allKeys));
+    const uniqueKeys = Array.from(new Set(allKeys))
+      .map((key) => Number(key))
+      .sort((a, b) => a - b)
+      .map((key) => key.toString());
```

Comment:
Can we do the converting to number in the sort function, so we can skip 1 conversion?

Reply:
(is it even needed at all?)

Reply:
I think a number conversion is correct as a lexical sort will give the wrong results in the case where timestamps are different number of digits (don't want to have to come back here and fix this in 2286 :grin:)

I guess I had convinced myself that it was more efficient to do the conversion once before sorting, instead of repeating the same conversion many time each time two numbers are compared. 

But given that the datasets are small and mostly in order anyway, I expect it is completely negligible.